### ISSUE TITLE
`mypy` error for `pd.concat(dict[str, Dataframe])`

### ISSUE THREAD

**Issue by janosh** [VirtusLab/pandas-stubs#149]
`pd.concat()` takes a mapping of both Series and Dataframe but is currently typed to accept only Series:

```py
import pandas as pd

df1 = pd.DataFrame([[1, 2, 3]], columns=["a", "b", "c"])
df1 = df1.copy()

dct = {"df1": df1, "df2": df1}
pd.concat(dct)
>>> mypy(error): Argument 1 to "concat" has incompatible type "Dict[str, DataFrame]";
    expected "Union[Sequence[Series], Mapping[Optional[Hashable], Series]]"
```

**Comment by zkrolikowski-vl** [comment]
Hi @janosh. Looks like it's not matching an appropriate declaration:
```def concat(objs: Union[Sequence[DataFrame], Mapping[Label, DataFrame]], axis: AxisOptionHorizontal = ..., join: str=..., ignore_index: bool=..., keys: Any = ..., levels: Any = ..., names: Any = ..., verify_integrity: bool=..., sort: bool=..., copy: bool=...) -> DataFrame: ...
```

I'll try to workout why this is happening.

**Comment by zkrolikowski-vl** [comment]
After some experimenting I've come to a conclusion that mypy exhibits a strange behaviour:
-  when passed `Mapping[Optional[Hashable], DataFrame]` as a variable it won't be able to find an appropriate overloaded declaration 
-  when passed the same value is  as an expression it works correctly

I'm forced to use `Any` instead of `Optional[Hashable]`. It allows for False Negatives but avoids False Positives in the spirit of: "You can't prove I'm wrong". I think that's a good tradeoff.

The tests included in https://github.com/VirtusLab/pandas-stubs/pull/153 should cover your use-case. Please have a look.

**Comment by janosh** [comment]
@zkrolikowski-vl Thanks. The new tests look good. 👍

**Comment by zkrolikowski-vl** [comment]
It's merged and version 1.2.0.51 on PyPI! Conda-forge will follow this afternoon.

Closing the issue.
